message DocIndexMessage
{
    required uint64 seq            = 1;
    required int64  nid            = 2;

    message Term {
        required uint64 term_sign  = 1;
        repeated uint32 occ_list   = 2;
    }

    message IndexField {
        required bytes field_name  = 1;
        repeated Term  terms       = 2;
    }

    message ProfileField {
        required bytes field_name  = 1;
        optional bytes field_value = 2;
    }

    repeated IndexField    index_fields     = 3;
    repeated ProfileField  profile_fields   = 4;
    required uint32        action           = 5;
}
